crates/bgp-pkt/src/wire/serializer/mod.rs

```diff
@@ -182,6 +182,37 @@ fn write_tlv_header<T: Write>(
     Ok(())
 }
 
+#[inline]
+/// Write a TLV header.
+/// ```text
+///  0                   1                   2                   3
+///  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |   Type        |             Length            |               ~
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// ```
+///
+/// `tlv_type` : tlv code point
+///
+/// `tlv_length` : total tlv length on the wire
+/// (as reported by the writer <=> including type and length fields)
+///
+/// Written length field will be `tlv_length - 3` since "Length" must not
+/// include the length of the "Type" and "Length" field
+fn write_tlv_header_t8_l16<T: Write>(
+    writer: &mut T,
+    tlv_type: u8,
+    tlv_length: u16,
+) -> Result<(), std::io::Error> {
+    // do not account for the tlv type u16 and tlv length u16
```

Comment:
you mean tlv type u8

Reply:
yep, stupid copy paste from the write_tlv_header_t16_l16 function :)